### PR TITLE
register downloaded non-base model

### PR DESCRIPTION
**Pull request by guangrui-cloudnatix**

### CODE REVIEW

File: api/v1/model_manager_service.proto [hunk 1]
```diff
@@ -114,13 +114,16 @@ enum QuantizationType {
 }
 
 message RegisterModelRequest {
+  string id = 7;
```
Comment (kkaneda) [inline api/v1/model_manager_service.proto:117]: Could you add a comment to describe how this field works?

File: server/internal/server/models.go [hunk 2]
```diff
@@ -204,16 +206,18 @@ func (s *WS) RegisterModel(
 		return nil, status.Error(codes.InvalidArgument, "project_id is required")
 	}
 
-	id, err := s.genenerateModelID(req.BaseModel, req.Suffix)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "generate model ID: %s", err)
+	id := req.Id
```
Comment (kkaneda) [inline server/internal/server/models.go:209]: Could you make this RPC return the `AlreadyExists` error if the `id` field is specified and there is already a model of the same ID?

File: loader/internal/loader/loader.go [hunk 3]
```diff
@@ -212,6 +254,48 @@ func (l *L) loadBaseModel(ctx context.Context, modelID string) error {
...
+
+	// HuggingFace uses '/" as a separator, but Ollama does not accept. Use '-' instead for now.
+	// TODO(kenji): Revisit this.
+	convertedModelID := strings.ReplaceAll(model.Model, "/", "-")
+
+	// First check if the model exists in the database.
+	ctx = auth.AppendWorkerAuthorization(ctx)
+	_, err := l.modelClient.GetModelPath(ctx, &mv1.GetModelPathRequest{Id: convertedModelID})
+	if err == nil {
+		l.log.Info("Model", convertedModelID, " exists. Do nothing.")
```
Comment (kkaneda) [inline loader/internal/loader/loader.go:270]: Let's follow the style of the structured logging. 

```suggestion
		l.log.Info("Already model exists", "modelID", convertedModelID)
```

File: loader/internal/loader/loader.go [hunk 4]
```diff
@@ -212,6 +254,48 @@ func (l *L) loadBaseModel(ctx context.Context, modelID string) error {
...
+	if _, _, err := l.downloadAndUploadModel(ctx, model.Model); err != nil {
+		return err
+	}
+
+	if _, err := l.modelClient.RegisterModel(ctx, &mv1.RegisterModelRequest{
+		Id:           convertedModelID,
+		BaseModel:    model.BaseModel,
+		Adapter:      v1.AdapterType(v1.AdapterType_value[model.AdapterType]),
+		Quantization: v1.QuantizationType(v1.QuantizationType_value[model.QuantizationType]),
+		// TODO(guangrui): Allow to configure project and org for models.
```
Comment (kkaneda) [inline loader/internal/loader/loader.go:286]: I see. Probably just allow users to set these in the configs.

**Review by kkaneda — APPROVED** [review]
LG!

File: server/internal/server/models.go (hunk 2 above)
Comment (guangrui-cloudnatix) [reply server/internal/server/models.go:209]: This field allows a user to name the model arbitrarily. The name of the model downloaded from HF can be in any format, we don't have to converted its name into "base-model-name.suffix" format.  

This ID would be also the model name to be used in inference request.

File: api/v1/model_manager_service.proto (hunk 1 above)
Comment (guangrui-cloudnatix) [reply api/v1/model_manager_service.proto:117]: This field allows a user to name the model arbitrarily. The name of the model downloaded from HF can be in any format, we don't have to converted its name into "base-model-name.suffix" format.  

This ID would be also the model name to be used in inference request.

Comment (kkaneda) [reply api/v1/model_manager_service.proto:117]: Ah yes, I understand. My comment was to add the comment in the code as the API contract is not very straightforwad.

File: api/v1/model_manager_service.proto [hunk 5]
```diff
@@ -114,13 +114,18 @@ enum QuantizationType {
 }
 
 message RegisterModelRequest {
+  // id is the model name.
+  string id = 7;
   string base_model = 1;
   // suffix is deprecated, use fine_tune and quantization instead.
```
Comment (kkaneda) [inline api/v1/model_manager_service.proto:120]: There is no `fine_tune` field?

**Pull request merged**